File: Modules/Auth/Models/DeviceSession.php
<?php

namespace Modules\Auth\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Modules\Auth\Enums\DeviceEventType;
use Modules\User\Models\User;

class DeviceSession extends Model
{
    public $timestamps = false;

    protected $fillable = [
        'user_id',
        'device_id',
        'device_info',
        'event_type',
        'sanctum_token_id',
        'created_at',
    ];

    protected $casts = [
        'event_type' => DeviceEventType::class,
        'device_info' => 'array',
        'created_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: Solid.Api/Database/Entities/DeviceSession.cs
namespace Solid.Api.Database.Entities;

public sealed class DeviceSession
{
    public long Id { get; set; }

    public long UserId { get; set; }

    public string DeviceId { get; set; } = string.Empty;

    public string? DeviceInfo { get; set; }

    public string EventType { get; set; } = string.Empty;

    public long? SanctumTokenId { get; set; }

    public DateTime CreatedAt { get; set; }

    public User User { get; set; } = null!;
}
